matrix/test_unittest_matrix.py

```diff
@@ -0,0 +1,181 @@
+"""Matrix class unit test.
+
+This module uses unittest and pytest to generate tests for the Matrix class
+from matrix.py. Aspects which are tested: initialization, immutability,
+mathematical operations, boolean operations, iterations and is hashable.
+"""
+
+import unittest
+
+from matrix import Matrix
+
+
+class TestMatrixCreate(unittest.TestCase):
+    """Test the creation of a matrix and that an instance can't be modified."""
+
+    def test_initialization(self):
+        """Test init function of Matrix."""
+        self.assertIsInstance(Matrix(((1, 2), (3, 4))), Matrix)
+        self.assertIsInstance(Matrix(((1, 2), (5.5, 4))), Matrix)
```

Comment:
This test is weird.
You create a new instance of Matrix, its very unlikely to return an instance that isn't matrix (you need to make effort to cause it).
What can happen, is the initialization fails of one of the reasons:
* You get exception in the init
* The instance doesn't represent what you wanted

Reply:
I didn't understand that, can you explain it in different words?

Reply:
If you create an object of type A, then if the init didn't failed, it will always return an object of type A, so checking it is meaningless
If you are checking that constructor works then there are two things you need to check:
The first is that it doesn't fail, e.g. you divide by zero in the  constructor, so it will throw an exception and fail
The second, it doesn't do what you want. For example, creating unity matrix of dim=1 need to create (1,) or ((1,)) (depend on how you implemented it), but it gave instead 1 (without the tuple)

Reply:
So I need to check if the constructor doesn't fail and that the return value is what I meant it would be?

Reply:
If you write a test for it, yes